core/src/main/java/org/infinispan/iteration/impl/TrackingEntryIterableFromStream.java

```diff
@@ -0,0 +1,98 @@
+package org.infinispan.iteration.impl;
+
+import org.infinispan.AdvancedCache;
+import org.infinispan.Cache;
+import org.infinispan.commons.util.CloseableIterable;
+import org.infinispan.commons.util.CloseableIterator;
+import org.infinispan.commons.util.Closeables;
+import org.infinispan.container.entries.CacheEntry;
+import org.infinispan.context.Flag;
+import org.infinispan.filter.CacheFilters;
+import org.infinispan.filter.Converter;
+import org.infinispan.filter.KeyValueFilter;
+import org.infinispan.filter.KeyValueFilterConverter;
+import org.infinispan.util.concurrent.ConcurrentHashSet;
+
+import java.util.EnumSet;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Stream;
+
+/**
+ * CloseableIterable that also tracks the iterators it spawns so it can properly close them when the close method is
+ * invoked.
+ *
+ * @author wburns
+ * @since 7.0
+ */
```

Comment:
Yeah I basically copied this from an existing 7.0 class as it does almost the same thing, heh.  Fixing and the javadoc comment.